apps/api/v2/src/modules/workflows/inputs/create-workflow.input.ts

```diff
@@ -0,0 +1,229 @@
+    required: false,
+  })
+  @IsOptional()
+  @IsString()
+  text?: string;
+}
+
+export class WorkflowStepDto {
+  @ApiProperty({ description: "Unique identifier of the step", example: 67244 })
+  @IsNumber()
+  id!: number;
+
+  @ApiProperty({ description: "Step number in the workflow sequence", example: 1 })
+  @IsNumber()
+  stepNumber!: number;
+
+  @ApiProperty({ description: "Action to perform", example: "EMAIL_HOST", enum: StepAction })
+  @IsEnum(StepAction)
+  action!: StepAction;
+
+  @ApiProperty({ description: "Recipient type", example: "ATTENDEE", enum: RecipientType })
+  @IsEnum(RecipientType)
+  recipient!: RecipientType;
+
+  @ApiPropertyOptional({
+    description: "Email address if recipient is EMAIL, required for action EMAIL_ADDRESS",
+    example: "31214",
+    required: false,
+    externalDocs: {
+      url: "https://cal.com/docs/api-reference/v2/organization-team-verified-resources/verify-an-email-for-an-org-team",
+    },
+  })
+  @ValidateIf((o) => Boolean(o.action && o.action === StepAction.EMAIL_ADDRESS))
+  @IsNumber()
+  verifiedEmailId?: number;
+
+  @ApiPropertyOptional({
+    description:
+      "Phone number if recipient is PHONE_NUMBER, required for actions SMS_NUMBER and WHATSAPP_NUMBER",
+    example: "3243434",
+    required: false,
+    externalDocs: {
+      url: "https://cal.com/docs/api-reference/v2/organization-team-verified-resources/verify-a-phone-number-for-an-org-team",
+    },
+  })
+  @ValidateIf((o) =>
+    Boolean(o.action && (o.action === StepAction.SMS_NUMBER || o.action === StepAction.WHATSAPP_NUMBER))
+  )
+  @IsNumber()
+  verifiedPhoneId?: number;
+
+  @ApiProperty({ description: "Template type for the step", example: "REMINDER", enum: TemplateType })
+  @IsEnum(TemplateType)
+  template!: TemplateType;
+
+  @ApiPropertyOptional({
+    description:
+      "Whether to include a calendar event in the notification, can be included with actions EMAIL_HOST, EMAIL_ATTENDEE, EMAIL_ADDRESS ",
+    example: true,
+  })
+  @ValidateIf((o) =>
+    Boolean(
+      o.action &&
+        (o.action === StepAction.EMAIL_HOST ||
+          o.action === StepAction.EMAIL_ADDRESS ||
+          o.action === StepAction.EMAIL_ATTENDEE)
+    )
+  )
+  @IsBoolean()
+  includeCalendarEvent = false;
+
+  @ApiProperty({ description: "Displayed sender name.", type: String })
+  @IsString()
+  sender!: string;
+
+  @ApiProperty({ description: "Message content for this step", type: WorkflowMessageDto })
+  @ValidateNested()
+  @Type(() => WorkflowMessageDto)
+  message!: WorkflowMessageDto;
+}
+
+export class CreateWorkflowDto {
+  @ApiProperty({ description: "Name of the workflow", example: "Platform Test Workflow" })
+  @IsString()
+  name!: string;
+
+  @ApiProperty({ description: "Activation settings for the workflow", type: WorkflowActivationDto })
+  @ValidateNested()
+  @Type(() => WorkflowActivationDto)
+  activation!: WorkflowActivationDto;
```

Comment:
I would split this file into multiple input files - 1 for activation dtos, another for triggers, another for steps this file is too long.